### PR TITLE
Huey main file changes

### PR DESCRIPTION
**Pull request by justinjosephmkj**
Made changes to main.py.

If trying to run 

` def execute_task(self, step: Dict[str, Any]) -> Any:`

 as a 

` @huey.task() ` then the code is not be able to call `execute_task()`

if the function `execute_task() `is defined as a normal function then the workflow will execute.

cd to the folder src, then use the following to run the code

`huey_consumer main.huey`

## Summary by Sourcery

Refactor main.py to enhance the cleanup process with error handling and logging, and remove unused code related to command-line argument parsing and Huey consumer startup. Adjust the execute_task function in emr_workflow.py by removing the Huey task decorator to ensure proper execution.

Enhancements:
- Improve cleanup process by adding error handling and logging for session manager closure.

Chores:
- Remove unused command-line argument parsing and Huey consumer startup code from main.py.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -110,93 +123,16 @@ def process_workflow(self) -> None:
 
 
 @huey.periodic_task(crontab(minute='*/5'))
-def schedule_tasks(config_file: str, workflow_config_file: str) -> None:
+def schedule_tasks() -> None:
```
[the remark this answers is not in the document]
Comment (yingbull) [reply src/main.py:126]: @justinjosephmkj  please respond to Sourcery.ai. @GetWellClinic

File: src/processors/o19/o19_updater.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			self.provider_number.append(category['default_tagger'])
+
+	system_type = self.config.get('emr.document_folder')
+
+	if system_type == 'pending':
+		return self.update_o19_pendingdocs(self)
+	else:
+		return self.update_o19_incomingdocs(self)
+
+def update_o19_pendingdocs(self):
```
Comment (yingbull) [inline src/processors/o19/o19_updater.py:45]: Lack of documentation, etc etc

File: src/processors/o19/o19_updater.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	response = self.session.post(url, data=params)
+
+	if response.status_code == 200:
+		return self.update_o19_last_processed_file(self)
+
+	self.logger.error(f"An error occurred: {response.status_code}")
+	return False
+
+
+def update_o19_incomingdocs(self):
```
Comment (yingbull) [inline src/processors/o19/o19_updater.py:78]: lack of documentation, etc etc, @justinjosephmkj @GetWellClinic

File: src/processors/patient_tagger/patient.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
+import re
+from bs4 import BeautifulSoup
+import itertools
+
+def get_patient_name(self):
```
Comment (yingbull) [inline src/processors/patient_tagger/patient.py:5]: Again, this level of lack of documentation is not something that should be considered done. @justinjosephmkj @GetWellClinic

File: src/processors/patient_tagger/patient.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        cleaned_string = text.replace("[", "").replace("]", "")
+        self.config.set_shared_state(type_of_query+'filter', cleaned_string)
+        return True,cleaned_string
+    else:
+        return False
+
+
+def unidentified_patients(self):
+    patient = '''{
+        "formattedDob": "2010-02-28",
```
Comment (yingbull) [inline src/processors/patient_tagger/patient.py:188]: What's this magic number?

File: src/processors/utils/ocr.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    Extract text from the provided PDF bytes using OCR.
+
+    :param pdf_bytes: The PDF content as a bytes-like object.
+    :return: True if text extraction is successful; False otherwise.
+    """
+    text = ''
+    try:
+        if self.enable_ocr_gpu:
+            self.logger.info("OCR using GPU")
+            device = torch.device("cuda:0")
```
Comment (yingbull) [inline src/processors/utils/ocr.py:64]: Should be configuration

File: src/auth/login_manager.py [hunk 7]
```diff
@@ -73,7 +73,7 @@ def login_with_selenium(self, driver):
         # Locate the login fields and enter credentials
         username_field = driver.find_element(By.NAME, "username")
         password_field = driver.find_element(By.NAME, "password")
-        pin_field = driver.find_element(By.NAME, "pin")
+        pin_field = driver.find_element(By.NAME, "pin2")
```
Comment (yingbull) [inline src/auth/login_manager.py:76]: pin2?

File: src/config/config_manager.py [hunk 8]
```diff
@@ -36,6 +50,18 @@ def get_workflow(self, key: str, default: Any = None) -> Any:
                 return default
         return value if value is not None else default
 
+    def update_lock_status(self, status: bool) -> None:
+        self.config['lock']['status'] = status
+        self.save_config()
+
+    def update_pending_inbox(self, file_name: str) -> None:
```
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply src/config/config_manager.py:137]: @yingbull Updated inline documentation. [https://github.com/GetWellClinic/ai-moa/pull/7/commits/ee18cc21c7273aecd746fa9c5b785bd696a26921](url)

File: src/processors/document_tagger/document_category.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
+
+
+def get_category_type(self):
```
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply src/processors/document_tagger/document_category.py:23]: @yingbull Updated inline documentation. [https://github.com/GetWellClinic/ai-moa/pull/7/commits/ee18cc21c7273aecd746fa9c5b785bd696a26921](url)

File: src/processors/o19/o19_updater.py (hunk 2 above)
Comment (justinjosephmkj) [reply src/processors/o19/o19_updater.py:72]: @yingbull Updated inline documentation. [https://github.com/GetWellClinic/ai-moa/pull/7/commits/ee18cc21c7273aecd746fa9c5b785bd696a26921](url)

File: src/processors/patient_tagger/patient.py (hunk 4 above)
Comment (justinjosephmkj) [reply src/processors/patient_tagger/patient.py:6]: @yingbull Updated inline documentation. [https://github.com/GetWellClinic/ai-moa/pull/7/commits/ee18cc21c7273aecd746fa9c5b785bd696a26921](url)

File: src/processors/patient_tagger/patient.py (hunk 5 above)
Comment (justinjosephmkj) [reply src/processors/patient_tagger/patient.py]: @yingbull It was DOB for unattached patient, moved to config file. [https://github.com/GetWellClinic/ai-moa/pull/7/commits/d7f33db255f3bd19ceae5b5af8278d30df79c6f8](url)

File: src/processors/utils/ocr.py (hunk 6 above)
Comment (justinjosephmkj) [reply src/processors/utils/ocr.py]: @yingbull Moved to config file. [https://github.com/GetWellClinic/ai-moa/pull/7/commits/d7f33db255f3bd19ceae5b5af8278d30df79c6f8](url)

File: src/auth/login_manager.py (hunk 7 above)
Comment (justinjosephmkj) [reply src/auth/login_manager.py:76]: @yingbull HTML element is identified using the element name 'pin2', the element pin is hidden.

**Review by yingbull — CHANGES_REQUESTED** [review]

File: config/template_providerlist.txt [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<report title="AI-MOA Config Search Providers (System generated)" description="List Providers for AI-MOA (v.2024.08.13)" active="1">
+
+<query>
+
+select last_name, first_name, provider_no, specialty, status from provider where status = 1 order by provider_no;
```
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply config/template_providerlist.txt:5]: @yingbull This is done through o19 Reports > Report by Template. The code uploads a template using Selenium and uses it to download the CSV, which will then be used in the code to identify providers from the file.

File: src/config/config_manager.py [hunk 11]
```diff
@@ -1,22 +1,100 @@
...
-        self.config = self.load_config(config_path)
-        self.workflow_config = self.load_config(workflow_config_path)
+        self.config_file = os.path.join(self.base_dir, config_file)
+        self.workflow_config_file = os.path.join(self.base_dir, workflow_config_file)
+        self.config = self.load_config(self.config_file)
+        self.workflow_config = self.load_config(self.workflow_config_file)
         self.in_memory_storage = {}
         self.shared_state = {}
 
+    def save_workflow_config(self) -> None:
```
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply src/config/config_manager.py:40]: @yingbull save_workflow_config() has documentation. please check https://github.com/GetWellClinic/ai-moa/blob/ee18cc21c7273aecd746fa9c5b785bd696a26921/src/config/config_manager.py#L40

File: src/config.yaml [hunk 12]
```diff
@@ -1,61 +1,112 @@
...
+ai:
+  uri: https://localhost:3334/v1/chat/completions  # URI endpoint for AI model interactions.
+  verify-HTTPS: false
 
-# Logging Configuration
-logging:
-  level: INFO
-  format: '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
-  filename: 'workflow.log'
+# AI-MOA document processor configuration. (Assumed to process medical-related documents.)
```
Comment (yingbull) [inline src/config.yaml:27]: No issue with AI generated docs but we do have to watch it for clarity; I would remove what is in brackets.

**Review by yingbull — APPROVED** [review]

File: src/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (GetWellClinic) [reply src/main.py:136]: Awaiting response from Justin.

File: src/config.yaml [hunk 13]
```diff
@@ -1,61 +1,71 @@
...
+    == ''assistant'' %}{{ message[''content''] + eos_token}}{% else %}{{ raise_exception(''Only
+    user and assistant roles are supported!'') }}{% endif %}{% endfor %}'
+  model: /models/model_name
+  temperature: 0.1
+  top_p: 0.1
+lock:
+  status: true
 logging:
-  level: INFO
+  filename: ../workflow.log
```
[the remark this answers is not in the document]
Comment (GetWellClinic) [reply src/config.yaml]: Awaiting response from Justin.

File: src/config/config_manager.py [hunk 14]
```diff
@@ -1,22 +1,100 @@
...
+
+    Attributes:
+        base_dir (str): The base directory where the config files are located.
+        config_file (str): Path to the general configuration file.
+        workflow_config_file (str): Path to the workflow configuration file.
+        config (dict): The loaded general configuration data.
+        workflow_config (dict): The loaded workflow configuration data.
+        in_memory_storage (dict): Temporary in-memory storage for data.
+        shared_state (dict): Shared state across different components of the application.
+    """
```
[the remark this answers is not in the document]
Comment (GetWellClinic) [reply src/config/config_manager.py:44]: Awaiting response from Justin.

File: src/config/config_manager.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (GetWellClinic) [reply src/config/config_manager.py:61]: Awaiting response from Justin.

File: src/config.yaml (hunk 12 above)
Comment (GetWellClinic) [reply src/config.yaml:27]: Awaiting response from Justin.

File: src/config.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply src/config.yaml]: @yingbull https://github.com/GetWellClinic/ai-moa/blob/1e096a61cd95ace06aa2354cb9c85f27daaafedf/src/config.yaml#L17

This is same as the base code, only updated the relative path.

File: src/config/config_manager.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply src/config/config_manager.py:44]: @yingbull https://github.com/GetWellClinic/ai-moa/blob/f7ca679650967534d7ba08231d8a9513f06d820a/src/main.py#L143 this is updated to environment variable, huey is not accepting arguments from cli

File: src/config/config_manager.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (justinjosephmkj) [reply src/config/config_manager.py:61]: @yingbull This was updated please see https://github.com/GetWellClinic/ai-moa/blob/f7ca679650967534d7ba08231d8a9513f06d820a/src/config/config_manager.py#L61

File: src/config.yaml (hunk 12 above)
Comment (justinjosephmkj) [reply src/config.yaml:27]: @yingbull https://github.com/GetWellClinic/ai-moa/blob/f7f6d676efab42c712e61022351ada0340eb5eae/src/config.yaml#L22 update please check. Thank you for letting me know.

**Pull request merged**